src/AuthenticationResponse.js

```diff
@@ -82,6 +102,37 @@ class AuthenticationResponse {
     return response
   }
 
+  /**
+   * errorResponse
+   *
+   * @param {AuthenticationResponse} response
+   *
+   * @throws {Error} If response params include the OAuth2 'error' param,
+   *   throws an error based on it.
+   *
+   * @returns {AuthenticationResponse} Chainable
+   *
+   * @todo Figure out HTTP status code (typically 400, 401 or 403)
+   *   based on the OAuth2/OIDC `error` code, probably using an external library
+   */
+  static errorResponse (response) {
+    const errorCode = response.params.error
+
+    if (errorCode) {
+      const errorParams = {}
+      errorParams['error'] = errorCode
+      errorParams['error_description'] = response.params['error_description']
+      errorParams['error_uri'] = response.params['error_uri']
+      errorParams['state'] = response.params['state']
```

Comment:
Also weird names, with underscore. What is state?

Reply:
Heh, yes, they are weird names, but they're straight from the [OAuth2 Error Response](https://tools.ietf.org/html/rfc6749#section-5.2) spec. And `state` is from the OIDC spec.